Mathlib/Computability/CFG.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Computability.Language
+
+/-!
+# Context-Free Grammar
+
+This file contains the definition of a Context-Free Grammar (CFG), which is a grammar that has a
+single nonterminal symbol on the left-hand side of each rule.
+Note that derivation by a grammar is inherently nondeterministic.
+-/
+
+/-- The type of symbols is the disjoint union of terminals and nonterminals. -/
+inductive Symbol (T : Type) (N : Type)
```

Comment:
Is there a reason you didn't make these universe polymorphic

Reply:
If I change everything to `Type _` then the final definition fails:
```
invalid occurrence of universe level 'u_2' at 'IsCF', it does not occur at the declaration type, nor it is explicit universe level provided by the user, occurring at expression
  Exists.{max (u_1 + 1) (u_2 + 2)} fun g ↦ Eq.{u_1 + 1} (CFG.language.{u_1, u_2} g) L
at declaration body
  fun {T : Type u_1} (L : Language T) ↦ ∃ (g : CFG T), CFG.language g = L
```

Reply:
Probably in that definition you will have to specify the definition as there exists a CFG in the same universe as the type of the language which gives the language